primitives/src/lib.rs

```diff
@@ -55,6 +89,13 @@ pub type CollectionId = u128;
 /// NFT Item ID
 pub type ItemId = u128;
 
+/// We assume that an on-initialize consumes 2.5% of the weight on average, hence a single extrinsic
+/// will not be allowed to consume more than `AvailableBlockRatio - 2.5%`.
+pub const AVERAGE_ON_INITIALIZE_RATIO: Perbill = Perbill::from_perthousand(25);
```

Comment:
these constants should be probably close to the payments pallet configs, is there some reason this should be in primitives?